project-base/storefront/components/Blocks/Product/LastVisitedProducts/LastVisitedProducts.tsx

```diff
@@ -0,0 +1,29 @@
+import { Heading } from 'components/Basic/Heading/Heading';
+import { ProductsSlider } from 'components/Blocks/Product/ProductsSlider';
+import { GetLastVisitedProductCatalogNumbers } from 'helpers/products/LastVisitedProducts';
+import { Webline } from 'components/Layout/Webline/Webline';
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { GtmProductListNameType } from 'types/gtm/enums';
+import { useProductsByCatnumsApi } from 'graphql/generated';
+
+const LastVisitedProducts: FC = () => {
+    const lastVisitedProductCatalogNumbers = GetLastVisitedProductCatalogNumbers();
+    const [result] = useProductsByCatnumsApi({ variables: { catnums: lastVisitedProductCatalogNumbers } });
```

Comment:
I don't see this query on server-side. This means it's only loaded on the client. I understand it might not be relevant to crawlers and robots, but the loading state should therefore be handled with a skeleton or similar.

Reply:
Hmmm... but when there will be no last visited... then it will jump from skeleton to "no content". I will discuss it with other developers

Reply:
I dont think its about discussing it with devs 🤔 Its a business decision. Either we want a loader or we dont. We are able to implement it gracefully.

Reply:
We have discussed it. First try was, let it without loader. Second was, lets save hasLastVisited:boolean to cookies. Then check and if it is true, show loader. This approach should avoid content jumping (;

Reply:
Fixed now we store info about last visited products in cookies again.